tools/clang/lib/SPIRV/SpirvEmitter.h

```diff
@@ -1156,6 +1156,13 @@ class SpirvEmitter : public ASTConsumer {
 
   /// The <result-id> of the OpString containing the main source file's path.
   SpirvString *mainSourceFile;
+
+  /// When there is an error, we want to stop compilation to prevent it from
+  /// causing segmentation faults. If it causes a segfault, we cannot check
+  /// the actual error message.
+  /// TODO: I am worried that this is a global variable in this class and it
+  ///       can be easily mis-used. Revisit this design.
+  bool stopEntireCompilation;
```

Comment:
The alternative here would be to flag that an error was found so that individual passes or handlers can query it and decide whether to return early.  Please file a feature request so we don't forget about this.

Reply:
Open in #2714